test/integration/reader-order.spec.ts

```diff
@@ -0,0 +1,184 @@
+
+// A mock reader that tracks when it's called and can be configured to succeed/fail
+class MockReader extends BarcodeReader {
+    static callOrder: string[] = [];
+    static shouldSucceed: Record<string, boolean> = {};
+    
+    FORMAT: string;
+    
+    constructor(config?: any, supplements?: Array<BarcodeReader>) {
+        super(config || {}, supplements);
+        this.FORMAT = config?.format || 'mock';
+    }
+    
+    decode(row?: Array<number>, start?: BarcodePosition | number): Barcode | null {
+        MockReader.callOrder.push(this.FORMAT);
+        
+        if (MockReader.shouldSucceed[this.FORMAT]) {
+            return {
+                code: `mock-${this.FORMAT}`,
+                start: 0,
+                end: 10,
+                startInfo: { start: 0, end: 3 },
+                format: this.FORMAT,
+            };
+        }
+        return null;
+    }
+    
+    static reset() {
+        MockReader.callOrder = [];
+        MockReader.shouldSucceed = {};
+    }
+}
+
+// Create specific mock reader classes for testing order
+class MockReaderFirst extends MockReader {
+    constructor(config?: any, supplements?: Array<BarcodeReader>) {
+        super({ ...config, format: 'mock_first' }, supplements);
+    }
+}
+
+class MockReaderSecond extends MockReader {
+    constructor(config?: any, supplements?: Array<BarcodeReader>) {
+        super({ ...config, format: 'mock_second' }, supplements);
+    }
+}
+
+class MockReaderThird extends MockReader {
+    constructor(config?: any, supplements?: Array<BarcodeReader>) {
+        super({ ...config, format: 'mock_third' }, supplements);
+    }
+}
+
+describe('Reader Order Tests', () => {
+    beforeEach(() => {
+        MockReader.reset();
+    });
+    
+    describe('Reader initialization order', () => {
+        it('should initialize readers in the order specified in config', () => {
+            // Register mock readers
+            Quagga.registerReader('mock_first_reader', MockReaderFirst);
+            Quagga.registerReader('mock_second_reader', MockReaderSecond);
+            Quagga.registerReader('mock_third_reader', MockReaderThird);
+            
+            // Create a decoder with readers in a specific order
+            const BarcodeDecoder = require('../../src/decoder/barcode_decoder').default;
+            
+            const config = {
+                readers: ['mock_first_reader', 'mock_second_reader', 'mock_third_reader'],
+                debug: {},
+            };
+            
+            // Create a minimal image wrapper mock
+            const mockImageWrapper = {
+                data: new Uint8Array(100),
+                size: { x: 10, y: 10 },
+                inImageWithBorder: () => true,
+            };
+            
+            const decoder = BarcodeDecoder.create(config, mockImageWrapper);
+            
+            // The decoder should have been created - readers are stored internally
+            expect(decoder).to.be.an('object');
```

Comment:
I don't understand how this tests the goal of testing that the readers are initialized in order.  shouldn't it test the callOrder somewhere to do that?  please explain or alter.